app/parsers/nmap.py

```diff
@@ -70,12 +71,15 @@ def parse_xml_report(self, root, name=None):
         return report
 
     def generate_network_map(self, report):
-        network_map = defaultdict(list)
-        report_hosts = report.hosts.keys()
-        for host in report_hosts:
-            if report.hosts[host].ports:
-                [network_map[h2].append(host) for h2 in report_hosts if h2 != host]
-        report.network_map = dict(network_map)
+        if report.network_map_nodes and report.network_map_edges:
+            return
+        network_map = nx.Graph()
+        for key,value in report.hosts.items():
```

Comment:
give more descriptive variable names for key and value

Reply:
Implementer preference, but if you want, it is a pythonism to use '_' for variables that you dont actually need reference to. 

For example, you use the `key` variable so you could do:

```
for _, value in report.hosts.items():
```

Reply:
I don't need the key from the .items(), soI went ahead and just changed to a .values() call to clean the code up.

Reply:
@heatonk - did you push that? I dont see it